### PR TITLE
Use appdirs to get the cache directory

### LINKED ISSUE #43: Use the XDG_CACHE_HOME environment variable if it exists
Following the discussion in #26, we need to make some changes to the `os_cache` function in [pooch/utils.py](https://github.com/fatiando/pooch/blob/master/pooch/utils.py) function to be compliant with the FreeDesktop standards:

* Add an option `check_environment` to the function that defaults to `False`
* If `check_environment == True`, check if the `XDG_CACHE_HOME` environment variable is defined.
* If it is, use it to define the default cache location instead of `~/.cache` on Linux.

### PR DESCRIPTION
**Pull request by remram44**
Rely on this is tiny module (no dependencies, one file, 600 lines) to do the right thing instead of implementing it ourselves. It follows the XDG specs for *NIX, but also correctly reads the location for Windows from the registry.

Adds `appdirs` as a dependency. Breaks compatibility by removing the `platform` argument of `os_cache` since it's no longer used. This should not be an issue since it was only there for testing purposes.

Fixes #43

### CODE REVIEW

File: pooch/utils.py [hunk 1]
```diff
@@ -40,15 +41,7 @@ def os_cache(project, platform=None):
...
-    if platform is None:
-        platform = sys.platform
-    if platform == "darwin":
-        cache_path = Path("~", "Library", "Caches", project)
-    elif platform == "win32":
-        cache_path = Path("~", "AppData", "Local", project, "cache")
-    else:  # *NIX
-        cache_path = Path("~", ".cache", project)
-    return cache_path
+    return appdirs.user_cache_dir(project)
```
Comment (remram44) [inline pooch/utils.py]: Should this be `Path(appdirs.user_cache_dir(...))`? The doctest seems to indicate so

**Comment by leouieda** [comment]
@remram44 I'll make some changes to fix the failing tests on this and merge it in. Thanks for all your input on this stuff!

**Comment by remram44** [comment]
You're very welcome!

File: pooch/utils.py (hunk 1 above)
Comment (leouieda) [reply pooch/utils.py]: It should. I changed that before merging.

**Comment by leouieda** [comment]
:+1: It's always good to get rid of code :)

**Pull request merged**